terraform/modules/firewall/main.tf

```diff
@@ -0,0 +1,172 @@
+locals {
+  common_tags = {
+    terraform = "true"
+  }
+  app = var.app
+  env = var.env
+}
+
+//TODO: 
+// - Add rate limiting rule to match BCDA's config
+// - SQL Injection ruleset
+// - Cloudfront distribution (?)
+resource "aws_wafv2_web_acl" "this" {
+  name  = "${var.app}-${var.env}-wafv2-web-acl"
+  scope = "REGIONAL"
+
+  default_action {
+    allow {
+
+    }
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "WAF_Common_Protections"
+    sampled_requests_enabled   = true
+  }
+
+  dynamic "rule" {
+    for_each = var.rate_based_rule != null ? [var.rate_based_rule] : []
```

Comment:
Let's try setting a rate-based rule here that we can generally use across WAFs instead of making it all dynamic. Base it on the BCDA rate limiting.